### PR TITLE
Fixed tests in recipe

### PR DESCRIPTION
**Pull request by nabobalis**

### CODE REVIEW

**Comment by Cadair** [comment]
I cancelled the appveyor build to prioritize the 0.8 release build. We can restart it later.

**Comment by Cadair** [comment]
Well the tests are still really failing in this build.

**Comment by Cadair** [comment]
@conda-forge/core is there a way to change the build so if the tests fail the build fails? This build faliure is indicative of a real problem with the package.

**Comment by msarahan** [comment]
I think conda-build should be failing when the tests fail.  If it's not, that might be the issue.  It could also be an issue with how conda-forge runs conda build, though (running the overall bash scripts without set -e ?)

**Comment by ericdill** [comment]
tl;dr

```yaml
   commands:
     - python -c "import sunpy, sys; sys.exit(sunpy.self_test())"
```

Full explanation:

The test suite is invoked like this:

```yaml
   commands:
     - python -c "import sunpy; sunpy.self_test()"
```

Which is running this from `sunpy/__init__.py`

```python
from sunpy.tests import main as self_test
```

which is, in turn calling a function that **returns** an exit code:

```python
def main(...):
    ....
    return pytest.main(all_args)
```

This has the net effect of being:

since `self_test()` is returning an exit code from the function you will need to wrap `sys.exit()` around `sunpy.self_test`, otherwise your test invocation will be effectively:

```bash
python -c "0"
```

which is not going to ever "fail"

This is what you'd need to have your test suite return the exit code from pytest which will trigger conda build to fail:

```python
import sunpy
import sys
ret_code = sunpy.self_test()
sys.exit(ret_code)
```

or, shorter:
```bash
python -c "import sunpy, sys; sys.exit(sunpy.self_test())"
```

**Comment by Cadair** [comment]
Thanks for the explanation @ericdill that's really useful.
